stdlib/public/Concurrency/YieldingContinuation.swift

```diff
@@ -0,0 +1,196 @@
+//===----------------------------------------------------------------------===//
+//
+// This source file is part of the Swift.org open source project
+//
+// Copyright (c) 2020-2021 Apple Inc. and the Swift project authors
+// Licensed under Apache License v2.0 with Runtime Library Exception
+//
+// See https://swift.org/LICENSE.txt for license information
+// See https://swift.org/CONTRIBUTORS.txt for the list of Swift project authors
+//
+//===----------------------------------------------------------------------===//
+
+import Swift
+
+@available(macOS 9999, iOS 9999, watchOS 9999, tvOS 9999, *)
+public struct YieldingContinuation<Element, Failure: Error>: Sendable {
```

Comment:
This type has reference semantics; is there any reason it's declared as a `struct`?

Reply:
originally we were intending to utilize mutations to enforce task isolation of the next function - which perhaps is still on the table